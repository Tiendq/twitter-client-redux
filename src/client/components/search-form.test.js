import React from 'react';
import { cleanup, fireEvent, render, renderIntoDocument } from 'react-testing-library';
import SearchForm from './search-form';

afterEach(() => cleanup());

test('should render correctly', () => {
  let ref = React.createRef();
  let { container } = render(<SearchForm startSearch={jest.fn()} forwardedRef={ref} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('search should start with "test"', () => {
  let ref = React.createRef();
  let startSearch = jest.fn();
  let { container } = renderIntoDocument(<SearchForm startSearch={startSearch} forwardedRef={ref} />);

  let keyword = 'test';

  let search = container.querySelector('input[type=text]');
  search.value = keyword;
  fireEvent.change(search);

  let submit = container.querySelector('button[type=submit]');
  fireEvent.click(submit);

  expect(startSearch).toBeCalledWith(keyword);
});

test('submit button should have correct classes', () => {
  let ref = React.createRef();
  let { container } = render(<SearchForm startSearch={jest.fn()} forwardedRef={ref} />);

  let submit = container.querySelector('button[type=submit]');
  expect(submit).toHaveClass('btn-success');
});
