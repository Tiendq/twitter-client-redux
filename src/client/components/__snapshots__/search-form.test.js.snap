// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<form
  class="col-12 searchForm"
>
  <div
    class="formGroup"
  >
    <input
      class="form-control"
      maxlength="50"
      placeholder="Search for a topic, full name, or @username"
      type="text"
      value=""
    />
    <button
      class="btn btn-success"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
